In [1]:
import seaborn as sns
import pandas as pd

<h3>Joint Probability</h3>

In [2]:
df = sns.load_dataset('titanic')
df

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
0           0       3    male  22.0      1      0   7.2500        S   Third   
1           1       1  female  38.0      1      0  71.2833        C   First   
2           1       3  female  26.0      0      0   7.9250        S   Third   
3           1       1  female  35.0      1      0  53.1000        S   First   
4           0       3    male  35.0      0      0   8.0500        S   Third   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
886         0       2    male  27.0      0      0  13.0000        S  Second   
887         1       1  female  19.0      0      0  30.0000        S   First   
888         0       3  female   NaN      1      2  23.4500        S   Third   
889         1       1    male  26.0      0      0  30.0000        C   First   
890         0       3    male  32.0      0      0   7.7500        Q   Third   

       who  adult_male deck  embark_town alive  alone  
0      man        True  NaN  Southampton    no  False  
1    woman       False    C    Cherbourg   yes  False  
2    woman       False  NaN  Southampton   yes   True  
3    woman       False    C  Southampton   yes  False  
4      man        True  NaN  Southampton    no   True  
..     ...         ...  ...          ...   ...    ...  
886    man        True  NaN  Southampton    no   True  
887  woman       False    B  Southampton   yes   True  
888  woman       False  NaN  Southampton    no  False  
889    man        True    C    Cherbourg   yes   True  
890    man        True  NaN   Queenstown    no   True  

[891 rows x 15 columns]

In [4]:
pd.crosstab(df['survived'],df['pclass'])

pclass      1   2    3
survived              
0          80  97  372
1         136  87  119

In [5]:
pd.crosstab(df['survived'],df['pclass'],normalize=True)

pclass           1         2         3
survived                              
0         0.089787  0.108866  0.417508
1         0.152637  0.097643  0.133558

<h4>0.089787, 0.108866,... are joint probabilities</h4>

<h3>Marginal probability</h3>

In [8]:
pd.crosstab(df['survived'],df['pclass'],normalize=True,margins=True)

pclass           1         2         3       All
survived                                        
0         0.089787  0.108866  0.417508  0.616162
1         0.152637  0.097643  0.133558  0.383838
All       0.242424  0.206510  0.551066  1.000000

<h4>the All column and row are the marginal probablities.<br><br>Whereas the normal row and column values except All row and column are joint probabilities.</h4>

<h3 style="color:red">Conditional Probability</h3>

<h4>It is the measure of probability of an event occuring given that another event has already occured. given by P(A|B) = P(A intersection B)/P(B)</h4>

In [10]:
pd.crosstab(df['survived'],df['pclass'],normalize="columns")

pclass          1         2         3
survived                             
0         0.37037  0.527174  0.757637
1         0.62963  0.472826  0.242363

<h4>The above results indicate that if a person is from pclass x then the probability of him dying/surviving is corresponding value.<br><br>Ex. if the person is from pclass 1 then probability of dying is 0.37037 and surviving is 0.62963. same for all pclasses.</h4>

In [11]:
pd.crosstab(df['survived'],df['pclass'],normalize="index")

pclass           1         2         3
survived                              
0         0.145719  0.176685  0.677596
1         0.397661  0.254386  0.347953

<h4>Now when "index" is given in normalize, it states that if the person is dying, then the probability of him being from pclass 1 is 0.145719 and from pclass 2 is 0.176685.<br><br> And if the person is surviving then the probability of him being from pclass 1 is 0.397661. and so on.</h4>

<h3 style="color:red">Baye's Theorem</h3>

<h3> P(A|B) = (P(B|A) . P(A)) / P(B)</h3>